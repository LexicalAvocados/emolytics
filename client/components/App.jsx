import React from 'react';
import { Route, Switch } from 'react-router-dom'
import { render } from 'react-dom';
import { createStore, applyMiddleware } from 'redux';
import { Provider } from 'react-redux';
import createHistory from 'history/createBrowserHistory';
import { ConnectedRouter, routerReducer, routerMiddleware } from 'react-router-redux';
import rootReducer from '../reducers/index.js';
import { changeExample } from '../actions';
import { connect } from 'react-redux';
import { bindActionCreators } from 'redux';
import * as ChangeActions from '../actions';

import DashboardHome from './creator/dashboard/dashboardHome.jsx';
import Navbar from './Navbar.jsx';
import Signup from './auth/signup.jsx';
import Login from './auth/login.jsx';
import TesterHome from './tester/TesterHome.jsx';
import TesterVideo from './tester/testerVideo.jsx';
import ProjectHome from './creator/project/projectHome.jsx';
import SectionHome from './creator/section/SectionHome.jsx';
import OptionHome from './creator/option/OptionHome.jsx';


class App extends React.Component {
  constructor(props) {
    super(props);
    this.state = {

    };
    this.onClick = this.onClick.bind(this);
    console.log('App: this:', this);
  }

  onClick() {
    this.props.actions.changeExample('changed');
  }

  render() {
    return (
      <div>
        <Navbar />
        {this.props.example.text}
        <button onClick={this.onClick}> test </button><br/><br/>

        {
          this.props.loggedInUser.username ? (
            this.props.loggedInUser.isCreator ? (
              <Switch>
                <Route exact path="/" component={DashboardHome}/>
                <Route path="/new" component={DashboardHome}/>
                <Route path="/testvideo" component={TesterVideo}/>
                <Route path="/project/:id" component={ProjectHome}/>
                <Route path="/section/:id" component={SectionHome}/>
                <Route path="/testviz" component={OptionHome}/>
                <Route path="/option/:id" component={OptionHome}/>
              </Switch>
            ) : (
              <Switch>
                <Route exact path="/" component={TesterHome}/>
                <Route path="/video/:id" component={TesterVideo}/>
              </Switch>
            )
          ) : (
            <Switch>
              <Route path="/signup" component={Signup}/>
              <Route path="*" component={Login}/>              
            </Switch>
          )
        }

      </div>
    )
  }
}

// React-Redux connect() boilerplate
// 1. Include the properties in the Store you want this component to have access to
// 2. Change the Component name at the very end to the one in the current file
const mapStateToProps = (state) => {
  console.log('state', state);
  return ({
    example: state.example,
    router: state.router,
    loggedInUser: state.loggedInUser
  })
};

const mapDispatchToProps = (dispatch) => ({
  actions: bindActionCreators(ChangeActions, dispatch)
});

export default connect(
  mapStateToProps,
  mapDispatchToProps
)(App);
