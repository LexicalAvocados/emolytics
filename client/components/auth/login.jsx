import React from 'react';
import { Form, FormGroup, FieldGroup, FormControl, ControlLabel, Checkbox, Button } from 'react-bootstrap';
import { Redirect } from 'react-router-dom';
import { connect } from 'react-redux';
import { store, bindActionCreators } from 'redux';
import * as ChangeActions from '../../actions';
import axios from 'axios';

class Login extends React.Component {
  constructor(props) {
    super(props);
    this.state = {
      typedUsername: '',
      typedPassword: '',
      loginError: null
    }
    this.updateTypedUsername = this.updateTypedUsername.bind(this);
    this.updateTypedPassword = this.updateTypedPassword.bind(this);
    this.submitLogin = this.submitLogin.bind(this);
    this.resetInputForms = this.resetInputForms.bind(this);
  }

  updateTypedUsername(e) {
    this.setState({typedUsername: e.target.value});
  }

  updateTypedPassword(e) {
    this.setState({typedPassword: e.target.value});
  }

  submitLogin(e) {
    e.preventDefault();
    axios.post('/login', {
      username: this.state.typedUsername,
      password: this.state.typedPassword
    })
      .then(res => {
        if (res.data.loggedIn) {
          this.props.actions.setLoggedIn(res.data.userData.username, res.data.userData.isCreator);
        } else {
          this.setState({loginError: res.data.reason});
          this.resetInputForms();
        }
      })
      .catch(err => {
        console.log('submitLogin Error:', err);
      });
  }

  resetInputForms() {
    this.setState({typedUsername: '', typedPassword: ''});
  }

  render() {
    return (
      <div>
        <Form horizontal onSubmit={this.submitLogin}>
          <FormGroup>
            {<div>this.state.loginError</div> && this.state.loginError}<br/><br/>
            <ControlLabel>Log In</ControlLabel><br/>
            <FormControl
              type='text'
              value={this.state.typedUsername}
              placeholder='Username'
              onChange={this.updateTypedUsername}
            /><br/>
            <FormControl
              type='password'
              value={this.state.typedPassword}
              placeholder='Password'
              onChange={this.updateTypedPassword}
            /><br/>
            <Button type='submit'>Submit</Button>
          </FormGroup>
        </Form>
      </div>
    )
  }
}

const mapStateToProps = (state) => {
  return ({
    example: state.example,
    setLoggedIn: state.setLoggedIn,
    router: state.router
  })
};

const mapDispatchToProps = (dispatch) => ({
  actions: bindActionCreators(ChangeActions, dispatch)
});

export default connect(
  mapStateToProps,
  mapDispatchToProps
)(Login);