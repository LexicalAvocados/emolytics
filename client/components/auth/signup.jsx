import React from 'react';
import { Form, FormGroup, FieldGroup, FormControl, ControlLabel, Checkbox, Button } from 'react-bootstrap';
import axios from 'axios';

// React-Redux connect() boilerplate
// NOTE: you may have to modify the filepath for ChangeActions
import { withRouter } from 'react-router-dom';
import { connect } from 'react-redux';
import { bindActionCreators } from 'redux';
import * as ChangeActions from '../../actions';

class Signup extends React.Component {
  constructor(props) {
    super(props);
    this.state = {
      typedUsername: '',
      typedPassword: '',
      typedEmail: '',
      isCreator: false
    }
    this.updateTypedUsername = this.updateTypedUsername.bind(this);
    this.updateTypedPassword = this.updateTypedPassword.bind(this);
    this.updateTypedEmail = this.updateTypedEmail.bind(this);
    this.updateIsCreator = this.updateIsCreator.bind(this);
    this.submitNewAccount = this.submitNewAccount.bind(this);
  }

  updateTypedUsername(e) {
    this.setState({typedUsername: e.target.value});
  }

  updateTypedPassword(e) {
    this.setState({typedPassword: e.target.value});
  }

  updateTypedEmail(e) {
    this.setState({typedEmail: e.target.value});
  }

  updateIsCreator(e) {
    this.setState({isCreator: !this.state.isCreator});
  }

  submitNewAccount(e) {
    e.preventDefault();
    axios.post('/signup', {
      username: this.state.typedUsername,
      password: this.state.typedPassword,
      email: this.state.typedEmail,
      isCreator: this.state.isCreator
    })
      .then(res => {
        this.props.actions.setLoggedIn(res.data.userData.username, res.data.userData.isCreator);
        this.props.history.push('/');
      })
      .catch(err => {
        console.log('submitNewAccount Error:', err);
      })
  }

  render() {
    return (
      <div>
        <Form horizontal onSubmit={this.submitNewAccount}>
          <FormGroup>
            <ControlLabel>Create New Account</ControlLabel><br/><br/>
            <FormControl
              type='text'
              value={this.state.typedUsername}
              placeholder='Username'
              onChange={this.updateTypedUsername}
            /><br/>
            <FormControl
              type='password'
              value={this.state.typedPassword}
              placeholder='Password'
              onChange={this.updateTypedPassword}
            /><br/>
            <FormControl
              type='text'
              value={this.state.typedEmail}
              placeholder='Email'
              onChange={this.updateTypedEmail}
            /><br/>
            <Checkbox onClick={this.updateIsCreator}>Register as a Creator</Checkbox><br/>
            <Button type='submit'>Submit</Button>
          </FormGroup>
        </Form>
      </div>
    )
  }
}

// React-Redux connect() boilerplate
// 1. Include the properties in the Store you want this component to have access to
// 2. Change the Component name at the very end to the one in the current file
const mapStateToProps = (state) => {
  return ({
    example: state.example,
    setLoggedIn: state.setLoggedIn,
    router: state.router
  })
};

const mapDispatchToProps = (dispatch) => ({
  actions: bindActionCreators(ChangeActions, dispatch)
});

export default withRouter(connect(
  mapStateToProps,
  mapDispatchToProps
)(Signup));