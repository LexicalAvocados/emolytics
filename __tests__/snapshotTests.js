import React from 'react';
import { MemoryRouter as Router } from 'react-router-dom';
import Enzyme from 'enzyme';
import Adapter from 'enzyme-adapter-react-15';
import { shallow } from 'enzyme';
import { App } from '../client/components/App.jsx';
import { Signup } from '../client/components/auth/signup.jsx';
import { Login } from '../client/components/auth/login.jsx';

Enzyme.configure({ adapter: new Adapter() });

// Run 'npm test' to run these tests. 
// These tests compare an Enzyme "Shallow Render" of the Component in question to an existing Snapshot taken by Jest.
// A Shallow Render only renders the Component in question, and does NOT render any child Components.
// If you make changes to any of these Components, or otherwise introduce code that causes a Shallow Render to not match a Snapshot,
// expect tests to break. Jest will show you the differences between the Shallow Render and the existing Snapshot.
// If you believe the new code to be sound,
// run 'npm test -- -u' to overwrite the existing Snapshot with a new Snapshot of the current Shallow Render.
// You can check existing snapshots in the __snapshots__ folder

// If you want to add similar tests for other components, follow these steps:
// 1. Add 'export' in front of 'class MyComponent extends React.Component' in the relevant Component file
// 2. Add 'import { MyComponent } from <MyComponent filepath>' to the top of this file
// (note: with the current setup we can only test the base Component Class, NOT the default export of the Component wrapped in withRouter + connect)
// 3. Follow the template seen below

describe('App Component Shallow Render', () => {

  it('renders accurately to the existing "App while logged in as Creator" snapshot', () => {
    const appCreator = shallow(<App loggedInUser={{username: 'a', isCreator: true}}/>);
    expect(appCreator).toMatchSnapshot();
  });

  it('renders accurately to the existing "App while logged in as Tester" snapshot', () => {
    const appTester = shallow(<App loggedInUser={{username: 'b', isCreator: false}}/>);
    expect(appTester).toMatchSnapshot();
  });

  it('renders accurately to the existing "App while not logged in" snapshot', () => {
    const appNoAuth = shallow(<App loggedInUser={{username: undefined, isCreator: undefined}}/>);
    expect(appNoAuth).toMatchSnapshot();
  });

});


describe('Signup Component Shallow Render', () => {
  
  it('renders accurately to the existing Signup snapshot', () => {
    const signup = shallow(<Signup />);
    expect(signup).toMatchSnapshot();
  });

});


describe('Login Component Shallow Render', () => {

  it('renders accurately to the existing Login snapshot', () => {
    const login = shallow(<Login />);
    expect(login).toMatchSnapshot();
  });

});