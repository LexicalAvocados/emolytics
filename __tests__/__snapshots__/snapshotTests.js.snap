// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App Component Shallow Render renders accurately to the existing "App while logged in as Creator" snapshot 1`] = `
<div>
  <withRouter(Connect(Navbar)) />
  <Switch>
    <Route
      component={[Function]}
      exact={true}
      path="/"
    />
    <Route
      component={[Function]}
      path="/new"
    />
    <Route
      component={[Function]}
      path="/testvideo"
    />
    <Route
      component={[Function]}
      path="/project:id"
    />
    <Route
      component={[Function]}
      path="/section:id"
    />
    <Route
      component={[Function]}
      path="/createProject"
    />
  </Switch>
</div>
`;

exports[`App Component Shallow Render renders accurately to the existing "App while logged in as Tester" snapshot 1`] = `
<div>
  <withRouter(Connect(Navbar)) />
  <Switch>
    <Route
      component={[Function]}
      exact={true}
      path="/"
    />
    <Route
      component={[Function]}
      path="/profile"
    />
    <Route
      component={[Function]}
      path="/video/:id"
    />
  </Switch>
</div>
`;

exports[`App Component Shallow Render renders accurately to the existing "App while not logged in" snapshot 1`] = `
<div>
  <withRouter(Connect(Navbar)) />
  <Switch>
    <Route
      component={[Function]}
      exact={true}
      path="/signup"
    />
    <Route
      component={[Function]}
      path="*"
    />
  </Switch>
</div>
`;

exports[`Login Component Shallow Render renders accurately to the existing Login snapshot 1`] = `
<div>
  <Form
    bsClass="form"
    componentClass="form"
    horizontal={true}
    inline={false}
    onSubmit={[Function]}
  >
    <FormGroup
      bsClass="form-group"
    >
      <br />
      <br />
      <ControlLabel
        bsClass="control-label"
        srOnly={false}
      >
        Log In
      </ControlLabel>
      <br />
      <Col
        bsClass="col"
        componentClass="div"
        sm={2}
      >
        <FormControl
          bsClass="form-control"
          componentClass="input"
          onChange={[Function]}
          placeholder="Username"
          type="text"
          value=""
        />
      </Col>
      <br />
      <br />
      <Col
        bsClass="col"
        componentClass="div"
        sm={2}
      >
        <FormControl
          bsClass="form-control"
          componentClass="input"
          onChange={[Function]}
          placeholder="Password"
          type="password"
          value=""
        />
      </Col>
      <br />
      <br />
      <Button
        active={false}
        block={false}
        bsClass="btn"
        bsStyle="default"
        disabled={false}
        type="submit"
      >
        Submit
      </Button>
    </FormGroup>
  </Form>
</div>
`;

exports[`Signup Component Shallow Render renders accurately to the existing Signup snapshot 1`] = `
<div>
  <Form
    bsClass="form"
    componentClass="form"
    horizontal={true}
    inline={false}
    onSubmit={[Function]}
  >
    <FormGroup
      bsClass="form-group"
    >
      <ControlLabel
        bsClass="control-label"
        srOnly={false}
      >
        Create New Account
      </ControlLabel>
      <br />
      <br />
      <Col
        bsClass="col"
        componentClass="div"
        sm={2}
      >
        <FormControl
          bsClass="form-control"
          componentClass="input"
          onChange={[Function]}
          placeholder="Username"
          type="text"
          value=""
        />
      </Col>
      <br />
      <br />
      <Col
        bsClass="col"
        componentClass="div"
        sm={2}
      >
        <FormControl
          bsClass="form-control"
          componentClass="input"
          onChange={[Function]}
          placeholder="Password"
          type="password"
          value=""
        />
      </Col>
      <br />
      <br />
      <Col
        bsClass="col"
        componentClass="div"
        sm={2}
      >
        <FormControl
          bsClass="form-control"
          componentClass="input"
          onChange={[Function]}
          placeholder="Email"
          type="text"
          value=""
        />
      </Col>
      <br />
      <br />
      <Checkbox
        bsClass="checkbox"
        disabled={false}
        inline={false}
        onClick={[Function]}
        title=""
      >
        Register as a Creator
      </Checkbox>
      <br />
      <br />
      <Button
        active={false}
        block={false}
        bsClass="btn"
        bsStyle="default"
        disabled={false}
        type="submit"
      >
        Submit
      </Button>
    </FormGroup>
  </Form>
</div>
`;
